vectorstores/alloydb/vectorstores_options.go

```diff
@@ -0,0 +1,128 @@
+package alloydb
+
+import (
+	"errors"
+
+	"github.com/tmc/langchaingo/embeddings"
+	"github.com/tmc/langchaingo/internal/alloydbutil"
+	"github.com/tmc/langchaingo/vectorstores"
+)
+
+const (
+	defaultSchemaName         = "public"
+	defaultIDColumn           = "langchain_id"
+	defaultContentColumn      = "content"
+	defaultEmbeddingColumn    = "embedding"
+	defaultMetadataJsonColumn = "langchain_metadata"
+	defaultK                  = 4
+)
+
+// AlloyDBVectoreStoresOption is a function for creating new vector store
+// with other than the default values.
+type AlloyDBVectoreStoresOption func(vs *VectorStore)
+
+// WithSchemaName sets the VectorStore's schemaName field.
+func WithSchemaName(schemaName string) AlloyDBVectoreStoresOption {
+	return func(v *VectorStore) {
+		v.schemaName = schemaName
+	}
+}
+
+// WithContentColumn sets VectorStore's the idColumn field.
+func WithIDColumn(idColumn string) AlloyDBVectoreStoresOption {
+	return func(v *VectorStore) {
+		v.idColumn = idColumn
+	}
+}
+
+// WithMetadataJsonColumn sets VectorStore's the metadataJsonColumn field.
+func WithMetadataJsonColumn(metadataJsonColumn string) AlloyDBVectoreStoresOption {
+	return func(v *VectorStore) {
+		v.metadataJsonColumn = metadataJsonColumn
+	}
+}
+
+// WithContentColumn sets the VectorStore's ContentColumn field.
+func WithContentColumn(contentColumn string) AlloyDBVectoreStoresOption {
+	return func(v *VectorStore) {
+		v.contentColumn = contentColumn
+	}
+}
+
+// WithEmbeddingColumn sets the EmbeddingColumn field.
+func WithEmbeddingColumn(embeddingColumn string) AlloyDBVectoreStoresOption {
+	return func(v *VectorStore) {
+		v.embeddingColumn = embeddingColumn
+	}
+}
+
+// WithMetadataColumns sets the VectorStore's MetadataColumns field.
+func WithMetadataColumns(metadataColumns []string) AlloyDBVectoreStoresOption {
+	return func(v *VectorStore) {
+		v.metadataColumns = metadataColumns
+	}
+}
+
+// WithOverwriteExisting is an option for VectorStore to
+// allow dangerous operations.
+func WithOverwriteExisting() AlloyDBVectoreStoresOption {
+	return func(v *VectorStore) {
+		v.overwriteExisting = true
+	}
+}
+
+// WithK sets the number of Documents to return from the VectorStore.
+func WithK(k int) AlloyDBVectoreStoresOption {
```

Comment:
Shouldn't this be a query option?

Reply:
Same as responded above, "K is being used on similaritySearch method, on python code is being passed on the vectorStore class as argument [here ](https://github.com/googleapis/langchain-google-cloud-sql-pg-python/blob/main/src/langchain_google_cloud_sql_pg/async_vectorstore.py#L59), it could be passed an argumet instead but then it would not satisfy the vectorstores.Vectorstore interface."